addons/hr_holidays/models/hr_leave.py

```diff
@@ -1151,20 +1152,24 @@ def _force_cancel(self, reason, msg_subtype='mail.mt_comment', notify_responsibl
                 responsibles = leave.employee_id.leave_manager_id.partner_id
                 responsibles |= leave.holiday_status_id.responsible_ids.partner_id
 
+            leave.sudo().state = "cancel"
```

Comment:
Why do you want to change the state here ? It was done before ar line 1173 :thinking:

Reply:
Thank you for the review! :smile: 
So the email template includes buttons/links to actions depending on the state of the leave, so you can for example refuse a time off request directly from the email. Which action link is included in the email is defined through `_notify_get_recipients_groups()` which gets called during the `leave.message_notify()`. So essentially, we want to set the state of the leave before we send the message so the correct/relevant actions are in the email/notification.